src/main/java/duke/task/Deadline.java

```diff
@@ -0,0 +1,70 @@
+package duke.task;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+
+/**
+ * Deadline task that inherits from Task class, and has an additional condition, which is when it is due by.
+ */
+public class Deadline extends Task {
+    protected LocalDate by;
+
+    /**
+     * Creates a new Deadline object
+     *
+     * @param description details about the Deadline
+     * @param by          date the deadline is due by in yyyy-mm-dd format
```

Comment:
there seems to be an awkward space in level 17? not sure if it is my computer issue